pkg/catalogv2/git/git.go

```diff
@@ -0,0 +1,480 @@
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/wrangler/pkg/randomtoken"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+	corev1 "k8s.io/api/core/v1"
+	k8snet "k8s.io/apimachinery/pkg/util/net"
+)
+
+type Options struct {
+	Credential        *corev1.Secret
+	CABundle          []byte
+	InsecureTLSVerify bool
+	Headers           map[string]string
+}
+
+func NewGit(directory, url string, opts *Options) (*Git, error) {
+	if opts == nil {
+		opts = &Options{}
+	}
+
+	g := &Git{
+		URL:               url,
+		Directory:         directory,
+		caBundle:          opts.CABundle,
+		insecureTLSVerify: opts.InsecureTLSVerify,
+		secret:            opts.Credential,
+		headers:           opts.Headers,
+	}
+	return g, g.setCredential(opts.Credential)
+}
+
+type Git struct {
```

Comment:
Since this is just being used in this package, does this struct need to be public?